src/main/java/com/epam/processor/DataProcessor.java

```diff
@@ -93,31 +151,53 @@ public RoadAccident getAccidentByIndex(String index){
      * @return
      */
     public Collection<RoadAccident> getAccidentsByLocation(float minLongitude, float maxLongitude, float minLatitude, float maxLatitude){
-        return null;
+        return roadAccidentList.stream()
+                .filter(roadAccident -> isGeoPositionInsideBoundaries(
+                        roadAccident.getLatitude(),
+                        roadAccident.getLongitude(),
+                        minLongitude, maxLongitude, minLatitude, maxLatitude))
+                .collect(toList());
+    }
+
+    private boolean isGeoPositionInsideBoundaries(float lat, float lon, float minLongitude, float maxLongitude, float minLatitude, float maxLatitude) {
+        return lat > minLatitude && lat < maxLatitude && lon > minLongitude && lon < maxLongitude;
     }
 
     /**
      * find the weather conditions which caused max number of incidents
      * @return
      */
     public List<String> getTopThreeWeatherCondition(){
-        return null;
+        Map<String, Long> weatherStatistics = roadAccidentList.stream()
+                .collect(groupingBy(RoadAccident::getWeatherConditions, counting()));
+
+        return weatherStatistics.entrySet().stream()
+                .sorted((o1, o2) -> o2.getValue().compareTo(o1.getValue()))
+                .map(stringLongEntry -> stringLongEntry.getKey())
+                .limit(3)
+                .collect(toList());
     }
 
     /**
      * count incidents by road surface conditions
      * @return
      */
     public Map<String, Long> getCountByRoadSurfaceCondition(){
-        return null;
+        return roadAccidentList.stream()
+                .collect(groupingBy(RoadAccident::getRoadSurfaceConditions, counting()));
     }
 
     /**
      * To match streaming operations result, return type is a java collection instead of multimap
      * @return
      */
     public Map<String, List<String>> getAccidentIdsGroupedByAuthority(){
-        return null;
+        return roadAccidentList.stream()
+                .collect(groupingBy(
+                        RoadAccident::getDistrictAuthority,
+                        HashMap::new,
```

Comment:
That's perfect solution from the first submition - very good. 
FYI - I did by myself without add HashMap::new - works fine as well.